### PR TITLE
staging-next 2022-04-10

### PR DESCRIPTION
**Pull request by vcunat**
I made this iteration a bit special, focused on security fixes and important bugfixes.  And perhaps some other changes that have very low breaking potential.  The purpose is to try making this iteration very fast, without requiring human work to progress, so that we don't block some notable problems:
- zlib update regressions:
  * https://github.com/NixOS/nixpkgs/issues/167708
  * https://github.com/NixOS/nixpkgs/pull/167395
  * https://github.com/NixOS/nixpkgs/issues/167115
- firefox addons: https://github.com/NixOS/nixpkgs/issues/167785
- - -
workflow docs: https://nixos.org/manual/nixpkgs/unstable/#submitting-changes-commit-policy
constitutents: https://hydra.nixos.org/job/nixpkgs/staging-next/unstable#tabs-constituents
jobset: https://hydra.nixos.org/jobset/nixpkgs/staging-next

Previous staging-next: https://github.com/NixOS/nixpkgs/pull/165406

### CODE REVIEW

**Comment by vcunat** [comment]
I somehow chose the changes by hand, selecting all corresponding PRs with "security" label and then passing through all the merges in `master..staging` and picking some more.  You can suggest changes to that as well, though we surely don't want to slow this iteration down too much.

The powerdns commit got left in `staging-next` by accident; it was intended for the previous iteration but got merged too late.  It could come directly to `master` now, but... now I didn't have sufficient motivation to act on it.

**Comment by K900** [comment]
https://github.com/NixOS/nixpkgs/pull/166335 should probably go in

**Comment by vcunat** [comment]
TL;DR: so that pycurl problem should be resolved now, both on master and also in a better way here on staging-next.

**Comment by fabianhjr** [comment]
>  made this iteration a bit special, focused on security fixes and important bugfixes.

>  You can suggest changes to that as well, though we surely don't want to slow this iteration down too much.

Hi @vcunat, just to clarify. The idea is to merge this before the scheduled breaking changes freeze of staging-staging next on April 17 and then open a new staging-next cycle then with the pending staging commits from this cycle?

**Comment by vcunat** [comment]
April 17 is the first time restrictions apply for changes merging **to** staging(-next) branch.  Commits that are there now certainly will reach 22.05.   (Well of course, it can happen that some of them will be found out to cause too much trouble and reverted, as usual.)

The release schedule itself even plans for two week-long staging(-next) iterations, though historically we typically struggled to make them that fast, so that part might happen a bit differently than planned.
